pkg/agent/knapsack/knapsack.go

```diff
@@ -43,42 +51,98 @@ func (k *Knapsack) getKVStore(storeType storage.Store) types.KVStore {
 	return store
 }
 
-func (k *Knapsack) AgentFlagsStore() types.KVStore {
+func (k *knapsack) AgentFlagsStore() types.KVStore {
 	return k.getKVStore(storage.AgentFlagsStore)
 }
 
-func (k *Knapsack) AutoupdateErrorsStore() types.KVStore {
+func (k *knapsack) AutoupdateErrorsStore() types.KVStore {
 	return k.getKVStore(storage.AutoupdateErrorsStore)
 }
 
-func (k *Knapsack) ConfigStore() types.KVStore {
+func (k *knapsack) ConfigStore() types.KVStore {
 	return k.getKVStore(storage.ConfigStore)
 }
 
-func (k *Knapsack) ControlStore() types.KVStore {
+func (k *knapsack) ControlStore() types.KVStore {
 	return k.getKVStore(storage.ControlStore)
 }
 
-func (k *Knapsack) InitialResultsStore() types.KVStore {
+func (k *knapsack) InitialResultsStore() types.KVStore {
 	return k.getKVStore(storage.InitialResultsStore)
 }
 
-func (k *Knapsack) ResultLogsStore() types.KVStore {
+func (k *knapsack) ResultLogsStore() types.KVStore {
 	return k.getKVStore(storage.ResultLogsStore)
 }
 
-func (k *Knapsack) OsqueryHistoryInstanceStore() types.KVStore {
+func (k *knapsack) OsqueryHistoryInstanceStore() types.KVStore {
 	return k.getKVStore(storage.OsqueryHistoryInstanceStore)
 }
 
-func (k *Knapsack) SentNotificationsStore() types.KVStore {
+func (k *knapsack) SentNotificationsStore() types.KVStore {
 	return k.getKVStore(storage.SentNotificationsStore)
 }
 
-func (k *Knapsack) StatusLogsStore() types.KVStore {
+func (k *knapsack) StatusLogsStore() types.KVStore {
 	return k.getKVStore(storage.StatusLogsStore)
 }
 
-func (k *Knapsack) ServerProvidedDataStore() types.KVStore {
+func (k *knapsack) ServerProvidedDataStore() types.KVStore {
 	return k.getKVStore(storage.ServerProvidedDataStore)
 }
+
+func (k *knapsack) RegisterChangeObserver(observer types.FlagsChangeObserver, flagKeys ...keys.FlagKey) {
+	k.flags.RegisterChangeObserver(observer, flagKeys...)
+}
+
+func (k *knapsack) SetDesktopEnabled(enabled bool) error {
+	return k.flags.SetDesktopEnabled(enabled)
+}
+func (k *knapsack) DesktopEnabled() bool {
+	return k.flags.DesktopEnabled()
+}
+
+func (k *knapsack) SetDebugServerData(debug bool) error {
+	return k.flags.SetDebugServerData(debug)
+}
+func (k *knapsack) DebugServerData() bool {
+	return k.flags.DebugServerData()
+}
+
+func (k *knapsack) SetForceControlSubsystems(force bool) error {
```

Comment:
The above funcs to retrieve KVStores are suffixed with `Store` should we consider suffixing the flags with `Flag`?

Reply:
I could do that if folks think it's more readable (or, conversely remove the `Store` suffix) from the ones above. Always a balance between readability and conciseness.